### PR TITLE
Adding the files needed to run the dashboard.

### PR DESCRIPTION
**Pull request by MilesDrabwell**
This adds the dashboard.py file that runs the streamlit dashboard and a notify.py file which contains functions used in the dashboard that allows users to sign up for AWS SNS email notifications.

### CODE REVIEW

**Comment by annac02** [comment]
Pylint is now above 9.8 and many bugs have been fixed, including changing colours for visibility. Any other improvements?

File: dashboard/dashboard.py [hunk 1]
```diff
@@ -0,0 +1,777 @@
...
+    return alt.layer(*graph).properties(
+        title=f"""Cases for Selected {filter_title.title()}s, Over Time""")
+
+
+def subscribe_to_court(courts: list):
+    """Allowing a SNS subscription to specific courts"""
+    st.header("Subscribe to Notifications")
+
+    email_input = st.text_input("Enter your email to subscribe:")
+    email = re.search(r"[\w_.-]+@[\w_.-]+[.]+[\w]+", email_input)
```
Comment (alabbahs) [inline dashboard/dashboard.py:574]: Definitely add the more robust regex that is on stack overflow to catch all the emails.

File: dashboard/dashboard.py [hunk 2]
```diff
@@ -0,0 +1,777 @@
...
+        st.markdown("""<span style='color:white'>Claimants</span>""",
+                    unsafe_allow_html=True)
+        prosecuting_participants = get_participants_and_lawyers_for_case(
+            conn, selected_case, False)
+        prosecuting_participants_html = format_participants_to_string(
+            prosecuting_participants)
+        st.markdown(prosecuting_participants_html,
+                    unsafe_allow_html=True)
+    with col2:
+        pass
```
Comment (alabbahs) [inline dashboard/dashboard.py:200]: What is this for?

File: dashboard/dashboard.py [hunk 3]
```diff
@@ -0,0 +1,777 @@
...
+
+
+def plot_filter_pie_tags(df: pd.DataFrame, selected_filter: list[str], field: str, tab: str,
+                         name: str):
+    """
+    Pie chart that displays the distribution of tags based on either the judges, courts or verdicts
+    """
+    filtered_data = df[df[tab].isin(selected_filter)]
+    aggregated_data = filtered_data.groupby(field).sum().reset_index()
+    aggregated_data = aggregated_data.sort_values(
```
Comment (alabbahs) [inline dashboard/dashboard.py:528]: DRY here

File: dashboard/dashboard.py [hunk 4]
```diff
@@ -0,0 +1,777 @@
...
+            sns_client = get_sns_client()
+            sub_to_topics(courts, sns_client, email.group())
+            st.success("Subscribed successfully!")
+        else:
+            st.error("Please enter a valid email address and select a judge.")
+
+
+def tabs():
+    """Separating our content into tabs"""
+    load_dotenv()
```
Comment (alabbahs) [inline dashboard/dashboard.py:591]: Why are you reloarding dotenv here?

File: dashboard/dashboard.py [hunk 5]
```diff
@@ -0,0 +1,777 @@
...
+            FROM court_case as cc
+            JOIN judge_assignment as ja ON ja.court_case_id = cc.court_case_id
+            JOIN judge as j ON j.judge_id = ja.judge_id
+            JOIN tag_assignment as ta ON ta.court_case_id = cc.court_case_id
+            JOIN tag as t ON t.tag_id = ta.tag_id
+            GROUP BY j.judge_name, t.tag_name;
+    """
+    with cnx.cursor(cursor_factory=RealDictCursor) as curs:
+        curs.execute(query)
+        result = curs.fetchall()
```
Comment (alabbahs) [inline dashboard/dashboard.py:392]: This is DRY on steroids, could replace these 3 lines every time with one function.

File: dashboard/README.md [hunk 6]
```diff
@@ -0,0 +1,38 @@
+# Court Transcript Dashboard
+
+## Viewing
+To view this dashboard, if not done so already, please make sure you have all the requirements installed. 
+This can be done by running `pip3 install -r requirements.txt` in the terminal; and is recommended to be done in a virtual environment.
+The dependencies include:
+```python
```
Comment (alabbahs) [inline dashboard/README.md:7]: maybe just add the commands for them to create the venv pip install the requirements in one line for ease.

**Review by alabbahs — APPROVED** [review]
Good work on the dashboard, it looks nice on the front, some DRYness in the back but overall great code. Amazing and clean implementation of SNS stuff. Gucci.

File: dashboard/dashboard.py (hunk 2 above)
Comment (MilesDrabwell) [reply dashboard/dashboard.py:202]: This is for the formatting of the dashboard, to make the spaces between graphs organised.

File: dashboard/dashboard.py (hunk 5 above)
Comment (MilesDrabwell) [reply dashboard/dashboard.py:394]: This is a good idea.

**Review by LinfanS — APPROVED** [review]
Thanks for making the requested changes, looks good now.

File: dashboard/.streamlit/config.toml [hunk 7]
```diff
@@ -0,0 +1,6 @@
+[theme]
```
[the remark this answers is not in the document]
Comment (MilesDrabwell) [reply dashboard/.streamlit/config.toml:1]: Done

**Pull request merged**